mirage/resolver/dns_resolver_mirage.ml

```diff
@@ -23,11 +23,24 @@ module Make (S : Tcpip.Stack.V4V6) = struct
         | x -> x
     end)
 
-  let resolver stack ?(root = false) ?(timer = 500) ?(udp = true) ?(tcp = true) ?tls ?(port = 53) ?(tls_port = 853) t =
+  type nonrec t = Dns_resolver.t ref
+
+  let connect (t : Dns_resolver.t) : t = { contents= t }
```

Comment:
I know what this code does, but I more often see people using the `ref` "constructor"
```suggestion
  let connect (t : Dns_resolver.t) : t = ref t
```